Multisig/Features/Security/KeychainStorage.swift

```diff
@@ -0,0 +1,240 @@
+//
+//  KeychainCenter.swift
+//  Multisig
+//
+//  Created by Dmitry Bespalov on 02.11.22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import CommonCrypto
+import LocalAuthentication
+
+class KeychainStorage {
+
+    static let sensitivePublicKeyTag = "global.safe.sensitive.public.key"
+    static let sensitiveEncryptedPrivateKeyTag = "global.safe.sensitive.private.key.as.encrypted.data"
+    static let derivedPasswordTag = "global.safe.password.as.data"
+    static let sensitiveKekTag = "global.safe.sensitive.KEK"
+
+    // used to store encrypted data as a password in keychain
+    static let defaultService: String = "encrypted_data"
+
+    // store passcode. Either if it random (user didn't give a password) or the user asked us to remember it
+    func storePasscode(derivedPasscode: String) throws {
+        // Store encrypted Password data
+        try storePasswordData(passwordData: derivedPasscode.data(using: .utf8)!)
+    }
+
+    func retrievePasscode() -> String? {
+        // Retrieve password from persistence
+        do {
+            if let passCodeData = try findPasswordData() {
+                return String.init(data: passCodeData, encoding: .utf8)
+            } else {
+                return nil
+            }
+        } catch {
+            return nil
+        }
+    }
+
+    private func storePasswordData(passwordData: Data) throws {
+        // delete existing sensitive key data
+        try deleteData(KeychainStorage.derivedPasswordTag)
+        // Create query
+        let addQuery = SQuery.generic(id: KeychainStorage.derivedPasswordTag, encryptedData: passwordData).queryData()
```

Comment:
I think merging them makes most sense. Otherwise we need to preface every call to store with a delete. Delete takes the query and store takes the item. See https://github.com/safe-global/safe-ios/pull/2792/commits/eb280ab2d2be7c272baa16640993ad8e0bb34675

Reply:
Done. See: 26f28f89e705e549ad4a149ba09a9baf9d16760d